tests/python_functional/functional_tests/source_drivers/network_source/proxyprotocol/test_pp_with_reconnect.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2020 One Identity
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+NUMBER_OF_MESSAGES = 3
+
+
+def test_pp_with_reconnenct(config, port_allocator, syslog_ng, loggen):
+    network_source = config.create_network_source(ip="localhost", port=port_allocator())
+    file_destination = config.create_file_destination(file_name="output.log")
+    config.create_logpath(statements=[network_source, file_destination])
+
+    syslog_ng.start(config)
+
+    loggen.start(
+        network_source.options["ip"], network_source.options["port"], inet=True,
+        rate=1, active_connections=1, reconnect=True, interval=30, number=NUMBER_OF_MESSAGES,
+    )
+
+    syslog_ng.stop()
+    syslog_ng.start(config)
+
+    logs = file_destination.read_logs(counter=NUMBER_OF_MESSAGES - 1)
+
+    # the send function of socket only write data to the output buffer, if the remote host is lost
+    # the message in the buffer will be discarded, result in len(logs)=NUMBER_OF_MESSAGES-1
+    assert len(logs) == NUMBER_OF_MESSAGES - 1
```

Comment:
I really appreciate, that you have written a `Light` test for this! Thanks!

However, this test does not use proxy-protocol, so this folder is not right for it. Thinking about it, it tests a `loggen` command, so technically a self-test should be written for it, but there is no place for that right now.
I think the best would be to move this to `functional_tests/source_drivers/network_source/` simply, then, when we make the self-tests for the `helper` classes, we can move it there, probably without using `syslog-ng`, not sure yet.

@mitzkia What do you think?

Reply:
Yes I also want to thank for updating Light (this is how we called framework under: `tests/python_functional/`) and adding new functional testcase.
Thanks for the comment, partially I am agreed with you. 
Basically this is a new loggen functional testcase for reconnect argument.
I would place it under `functional_tests/tools/loggen/test_loggen_with_reconnect.py`

And just for the note, at the end please add new testcase file to the list of `tests/python_functional/functional_tests/Makefile.am`

Reply:
Thanks for your comments, I would move it to `functional_tests/tools/loggen/test_loggen_with_reconnect.py`.

Reply:
Thanks!